lib/cstruct_cap.mli

```diff
@@ -0,0 +1,85 @@
+type 'a rd = < rd: unit; .. > as 'a
+type 'a wr = < wr: unit; .. > as 'a
+
+type 'a t
+type buffer = (char, Bigarray.int8_unsigned_elt, Bigarray.c_layout) Bigarray.Array1.t
+
+type rdwr =  < rd: unit; wr: unit; >
+type ro = < rd: unit; >
+type wo = < wr: unit; >
+
+type uint8 = int
+type uint16 = int
+type uint32 = int32
+type uint64 = int64
+
+val create : int -> rdwr t
+val create_unsafe : int -> rdwr t
+
+val ro : 'a rd t -> ro t
+val wo : 'a wr t -> wo t
+
+val of_string : ?off:int -> ?len:int -> string -> rdwr t
+val of_bytes : ?off:int -> ?len:int -> bytes -> rdwr t
+val of_hex : string -> rdwr t
+
+val to_bigarray : 'a t -> buffer
+
+val equal : 'a rd t -> 'b rd t -> bool
+val compare : 'a rd t -> 'b rd t -> int
+
+val check_bounds : 'a rd t -> int -> bool
+val check_alignment : 'a rd t -> int -> bool
+
+val get_char : 'a rd t -> int -> char
+val get_uint8 : 'a rd t -> int -> uint8
+val set_char : 'a wr t -> int -> char -> unit
+val set_uint8 : 'a wr t -> int -> uint8 -> unit
+
+val sub : 'a rd t -> int -> int -> 'a rd t
+val shift : 'a rd t -> int -> 'a rd t
+val to_string : ?off:int -> ?len:int -> 'a rd t -> string
+val to_bytes : ?off:int -> ?len:int -> 'a rd t -> bytes
+
+val blit : 'a rd t -> int -> 'b wr t -> int -> int -> unit
+val blit_from_string : string -> int -> 'a wr t -> int -> int -> unit
+val blit_from_bytes : bytes -> int -> 'a wr t -> int -> int -> unit
+val blit_to_bytes : 'a rd t -> int -> bytes -> int -> int -> unit
```

Comment:
labels would be nice in the above block of `blit` functions -- `src` `src_off` `dst` `dst_off` `length`